### DISCUSSION TITLE
inline style ``color:transparent`` in ``Image``

### DISCUSSION THREAD

**Discussion by 0xdbe** [vercel/next.js#61209] [Help] [answered]
### Summary

App [template](https://github.com/vercel/next.js/blob/canary/packages/create-next-app/templates/app-tw/ts/app/page.tsx) used by ``create-next-app``, include Next logo in ``app/page.tsx`` :

```
<Image
    className="relative dark:drop-shadow-[0_0_0.3rem_#ffffff70] dark:invert"
    src="/next.svg"
    alt="Next.js Logo"
    width={180}
    height={37}
    priority
  />
```

This image is translate by the following content:
```
<img 
  alt="Next.js Logo"
  fetchPriority="high" 
  width="180" 
  height="37" 
  decoding="async" data-nimg="1" 
  class="relative dark:drop-shadow-[0_0_0.3rem_#ffffff70] dark:invert" 
  style="color:transparent" 
  src="/next.svg"
/>
```

inline style is not possible with a strict Content Security Policy.

Question: why ``color:transparent`` is set via inline style ? is it because of Tailwind or ``next/image`` component ?

### Example

https://vercel-gtm.vercel.app/

**Comment by icyJoseph** [comment]
I guess, the real question, is whether or not you can use style at all, because for example setting `fill` to true, does inline other things as well. 

This comes from next/image though, it is an effort to hide the ALT text IIRC, when it is not yet necessary. Although the next/image works well with JS disabled, when it is enabled there's some calculations going on that, imply tinkering with the alt text visibility.

**Reply by 0xdbe** [reply]
@icyJoseph: thank you for this information. I understand now that ``style`` attribute come from ``next/image``.

To answer the real question: 
> whether or not you can use style at all

For secure website ``<style>`` tag and ``style=""`` attribute can't be used anymore. 
Otherwise, the Content-Security-Policy should be configured with  ``unsafe-inline`` for ``script-src`` directive.

In order to find a fix, I try to remove ``alt`` attribute, but it is not possible:

```
Property 'alt' is missing in type
```

**Reply by icyJoseph** [reply]
Yeah I know that CSP makes inline style a pain to work with. 

I am sorry you misunderstood that you had any control over the alt text, showing or hiding, behavior. You don't.

As said, this is but a use case, fill mode would even put more inline styles.

Perhaps some code changes are required to the Image component, in order to support this use case.

Alternatively, if you are using next 14.1, you have access to `getImageProps` which gives you the behavior of next/image as props. There I guess you could drop the style prop. I haven't had time to test it! 

- https://nextjs.org/blog/next-14-1#nextimage-support-for-picture-and-art-direction

**Reply by 0xdbe** [reply]
I took some time to test the workaround, without success. `style` attribute is always there.
A fix in the image component will be a big security improvement.
It could be possible load image style `color:transparent` from a class.

-----

## Test 1

```
const {
    props: { srcSet: src, ...rest },
  } = getImageProps({ src: "/vercel.svg", alt: 'Vercel Logo', width: 100, height: 24, style: { background: 'blue' } });
...
<img {...rest} className="dark:invert" />
```

Rendering:
```
<img alt="Vercel Logo" loading="lazy" width="200" height="24" decoding="async" style="color:transparent" src="/vercel.svg" class="dark:invert">
```

## Test 2

```
  const {
    props: { srcSet: src, ...rest },
  } = getImageProps({ src: "/vercel.svg", alt: 'Vercel Logo', width: 200, height: 24, style: { background: 'blue' } });
```

Rendering:
```
<img alt="Vercel Logo" loading="lazy" width="200" height="24" decoding="async" style="color:transparent;background:blue" src="/vercel.svg" class="dark:invert">
```

**Reply by icyJoseph** [reply, marked as answer]
Hi,

Those are not very good tests... you should've done this:

```jsx
import { getImageProps } from 'next/image';

export default function Home() {
  const { props } = getImageProps({
    src: '/next.svg',
    alt: 'Vercel',
    width: 100,
    height: 50,
  });

  const { style: _omit, ...delegated } = props;
  return <img {...delegated} />;
}

```

See it live here, https://stackblitz.com/edit/nextjs-gvqkk4?file=package.json,app%2Fpage.tsx,app%2Flayout.tsx

**Reply by icyJoseph** [reply]
Have you searched the Issues, perhaps someone has already reported this, and gotten a workaround?

**Reply by 0xdbe** [reply]
it finally works with your code snippet.
But this is a workaround in waiting a fix in next/image component.

Thanks, @icyJoseph

**Reply by 0xdbe** [reply]
Issue link to this discussion https://github.com/vercel/next.js/issues/61388

**Reply by icyJoseph** [reply]
Hi cool!, I also found this old issue: https://github.com/vercel/next.js/issues/45184 left a comment there
